keps/sig-storage/3333-retroactive-default-storage-class/README.md

```diff
@@ -393,7 +393,161 @@ will indicate a problem with the feature.
+
+```
+
+**Downgrade cluster**
+
+```bash
+$ yum history | grep -E "kubeadm|kubelet"
+    10 | install -y kubelet-1.25.2-0                                                                                                                                                                                                                                       | 2022-10-12 11:06 | Upgrade        |    1   
+     8 | install -y kubeadm-1.25.2-0                                                                                                                                                                                                                                       | 2022-10-12 09:45 | Upgrade        |    1   
+     7 | install -y kubelet-1.24.5-0 kubeadm-1.24.5-0 kubectl
+
+$ sudo yum -y history undo 8 && sudo yum -y history undo 10
+```
+
+**Perform post-rollback tests**
+
+Remove default SC:
+```bash
+$ kc patch sc/csi-hostpath-sc -p '{"metadata":{"annotations":{"storageclass.kubernetes.io/is-default-class":"false"}}}'
+storageclass.storage.k8s.io/csi-hostpath-sc patched
+```
+
+Create new PVC without SC:
+```yaml
+apiVersion: v1
+kind: PersistentVolumeClaim
+metadata:
+  name: csi-pvc-2
+spec:
+  accessModes:
+  - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+```
+
+```bash
+$ kc get pvc
+NAME        STATUS    VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
+csi-pvc     Bound     pvc-06a964ca-f997-4780-8627-b5c3bf5a87d8   1Gi        RWO            csi-hostpath-sc   96m
+csi-pvc-2   Pending     
+```
+
+Add default SC again:
+```bash
+$ kc patch sc/csi-hostpath-sc -p '{"metadata":{"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}'
+storageclass.storage.k8s.io/csi-hostpath-sc patched
+```
+
+Verify that the new PVC did not get updated with SC this time:
+```bash
+$ kc get pvc/csi-pvc-2
+NAME        STATUS    VOLUME   CAPACITY   ACCESS MODES   STORAGECLASS   AGE
+csi-pvc-2   Pending    
+```
+
+**Upgrade cluster again**
+
+Install/update kubeadm:
+```bash
+$ sudo yum install -y kubeadm-1.25.2-0
+```
+
+Perform kubeadm upgrade:
+```bash
+$ sudo kubeadm upgrade plan --config /mnt/clusterconf-examples/featuregate.yaml
+$ sudo kubeadm upgrade apply --config /mnt/clusterconf-examples/featuregate.yaml v1.25.2
+```
+
+Perform kubelet upgrade:
+```bash
+$ sudo yum install -y kubelet-1.25.2-0
+$ sudo systemctl daemon-reload 
+$ sudo systemctl restart kubelet
+```
+
+**Perform post-upgrade tests again**
+
+Verify that PVC got SC assigned right after upgrade and PV was provisioned and bound:
+```bash
+$ kc get pvc
+NAME        STATUS   VOLUME                                     CAPACITY   ACCESS MODES   STORAGECLASS      AGE
+csi-pvc     Bound    pvc-06a964ca-f997-4780-8627-b5c3bf5a87d8   1Gi        RWO            csi-hostpath-sc   117m
+csi-pvc-2   Bound    pvc-2e765394-f32c-42fb-b3db-ffe203612bac   1Gi        RWO            csi-hostpath-sc   24m
+```
+
```

Comment:
Somehow the second part of my comment was lost

L569: was the metric added? Can you please link its name?

Reply:
@wojtek-t The metric names match with what we have here and were added in https://github.com/kubernetes/kubernetes/pull/113323

Reply:
@wojtek-t  I'm not sure I follow, the names are written in `Monitoring requirements` section:

```
By inspecting a `retroactive_storageclass_total` metric value. If the counter
is increasing while letting PVCs being updated retroactively with a default 
StorageClass the feature is enabled. And at the same time if
`retroactive_storageclass_errors_total` counter does not increase the feature
works as expected.
```

Should that go elsewhere maybe?

Reply:
Lines 560-561 currently say:
```
A counter metric will be present in KCM metric endpoint, it will show total
count of successful and failed retroactive StorageClass assignments.
```

I'm only asking to put metrics names there explicitly too.